src/Domain/ValueObjects/WalletValueObject.cs

```diff
@@ -0,0 +1,43 @@
+using Domain.Aggregates;
+
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance)
+{
+    public User User { get; set; }
+    private decimal _balance = default;
+
+    public decimal Balance
```

Comment:
this good, encapsulation but validation should be handled inside the trytransfer method, which returns booleans.

here is what i suggest you could do instead:
```cs
record Wallet(decimal Balance)
{
  public decimal Balance { get; private set; } = Balance;

  public bool TryTransfer(Wallet other, decimal balance)
  {
    // do the validation here, and return bool if we cant transfer the funds.
    // this is a better approach than using exceptions because the consuming code will be cleaner - try-catch-free
  }
}